### PR TITLE
GCI-1022: Add certified copy item to basket

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Functionality broadened so that either a certificate or a certified copy item may be added to the basket, and the details of the item viewed in the response to a get basket request.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/service/ApiClientService.java [hunk 1]
```diff
@@ -19,28 +19,41 @@
...
+        final BaseItemApi baseItemApi;
+        try {
+            baseItemApi = apiClient
+                    .getInternalApiClient()
+                    .privateItemResourceHandler()
+                    .getItem(itemUri)
+                    .execute()
+                    .getData();
+        } catch (URIValidationException uve) {
+            throw new ServiceException("Unrecognised uri pattern for " + itemUri);
```
Comment (rjadhav-ch) [inline src/main/java/uk/gov/companieshouse/orders/api/service/ApiClientService.java:50]: Does this message cover invalid items?

File: src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToOrderMapperTest.java [hunk 2]
```diff
@@ -180,4 +180,6 @@ void testCheckoutToOrderMapping() {
         assertThat(order.getData().getOrderedBy(), is(checkout.getData().getCheckedOutBy()));
     }
 
+    // TODO GCI-984 Implement test for certified copy mapping?
```
Comment (rjadhav-ch) [inline src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToOrderMapperTest.java:183]: Are we okay to leave TODOs in `master`?

File: src/main/java/uk/gov/companieshouse/orders/api/service/ApiClientService.java (hunk 1 above)
Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/service/ApiClientService.java:50]: Yes, that's correct. This exception will be thrown should the URI not find a match in both of the patterns we currently have in the SDK for items.

File: src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToOrderMapperTest.java (hunk 2 above)
Comment (VaughanJackson123) [reply src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToOrderMapperTest.java:183]: Good question. I could not find any mention of TODOs in this guide: https://github.com/companieshouse/styleguides/blob/master/java.md. However, I do remember someone telling me that TODOs were OK so long as the JIRA that would deal with them was clearly identified (eg `TODO GCI-984`).

@sroberts2 Do you know whether there is a prohibition on TODOs going in to `master`?

Comment (sroberts2) [reply src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToOrderMapperTest.java:183]: This is fine as long as it has a Jira ticket and that we do eventually plan to address it.

File: src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToPaymentDetailsMapperTest.java [hunk 3]
```diff
@@ -148,6 +148,8 @@ public void testCheckoutToPaymentDetailsMapper() {
         testItems(source, target);
     }
 
+    // TODO GCI-984 Implement test for certified copy mapping?
```
[the remark this answers is not in the document]
Comment (VaughanJackson123) [reply src/test/java/uk/gov/companieshouse/orders/api/mapper/CheckoutToPaymentDetailsMapperTest.java:151]: I refer you to the above comment https://github.com/companieshouse/orders.api.ch.gov.uk/pull/64#discussion_r456299194.

**Comment by VaughanJackson123** [comment]
As a result of initial tests for GCI-984, I have discovered that these changes for GCI-1022 will introduce a bug in existing functionality. They cause the loss of specific item options information from the database when the order is paid for. I think it's best not to introduce that to master, even for a day or two. So I will take down this PR for now and see whether I can tackle that issue before rising another PR for these changes.

Hence **closing** this PR.

**Pull request closed without merging**